ios/RNDeviceInfo/RNDeviceInfo.m

```diff
@@ -819,6 +819,14 @@ - (NSDictionary *) getAvailableLocationProviders {
     resolve([EnvironmentValues objectAtIndex:[EnvironmentUtil currentAppEnvironment]]);
 }
 
+RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD(getBrightnessSync) {
+    return @([UIScreen mainScreen].brightness);
+}
+
+RCT_EXPORT_METHOD(getBrightness:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject) {
+    resolve(@([UIScreen mainScreen].brightness));
+}
+
```

Comment:
I just checked and this symbol does not seem to be available on tvOS so this will break some people's compilation: https://developer.apple.com/documentation/uikit/uiscreen/1617830-brightness

Also, the method implementation is duplicated when the async one can just refer to the sync one, so we can do the tvOS ifdef once:

```suggestion
- (float) getBrightness {
#if !TARGET_OS_TV
    return @([UIScreen mainScreen].brightness);
#else
    return [@1 floatValue];
#endif
}

RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD(getBrightnessSync) {
    return @(self.getBrightness);
}

RCT_EXPORT_METHOD(getBrightness:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject) {
    resolve(@(self.getBrightness));
}

```

what do you think of this refactor / tvOS compat suggestion? Does it still work for you like this? (I just coded it in the web ui here so I'm unsure...)

Reply:
Yes, I agree with this suggestion

Reply:
:-) - my new years resolution is "support all the platforms", and though I'm not there yet - it's a process - it means I'm keenly aware right now of all the little common failures in the build matrix...